src/main/java/org/retest/rebazer/connector/GithubConnector.java

```diff
@@ -0,0 +1,136 @@
+		return !getLastCommonCommitId( pullRequest ).equals( getHeadOfBranch( pullRequest ) );
+	}
+
+	String getHeadOfBranch( final PullRequest pullRequest ) {
+		return jsonPathForPath( "/git/refs/heads/" + pullRequest.getDestination() ).read( "$.object.sha" );
+	}
+
+	String getLastCommonCommitId( final PullRequest pullRequest ) {
+		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) + "/commits" );
+
+		final List<String> commitIds = jsonPath.read( "$..sha" );
+		final List<String> parentIds = jsonPath.read( "$..parents..sha" );
+
+		return parentIds.stream().filter( parent -> commitIds.contains( parent ) ).findFirst()
+				.orElseThrow( IllegalStateException::new );
+	}
+
+	@Override
+	public void merge( final PullRequest pullRequest ) {
+		final String message = String.format( "Merged in %s (pull request #%d) by ReBaZer", pullRequest.getSource(),
+				pullRequest.getId() );
+		final Map<String, String> request = new HashMap<>();
+		request.put( "commit_title", message );
+		request.put( "merge_method", "merge" );
+
+		template.put( requestPath( pullRequest ) + "/merge", request, Object.class );
+	}
+
+	@Override
+	public boolean greenBuildExists( final PullRequest pullRequest ) {
+		final String urlPath = "/commits/" + pullRequest.getSource() + "/status";
+		final DocumentContext jsonPath = jsonPathForPath( urlPath );
+		return jsonPath.<List<String>> read( "$.statuses[*].state" ).stream().anyMatch( s -> s.equals( "success" ) );
+	}
+
+	@Override
+	public List<PullRequest> getAllPullRequests() {
+		final DocumentContext jsonPath = jsonPathForPath( "/pulls" );
+		return parsePullRequestsJson( jsonPath );
+	}
+
+	public static List<PullRequest> parsePullRequestsJson( final DocumentContext jsonPath ) {
+		final List<Integer> pullRequestAmount = jsonPath.read( "$..number" );
+		final int numPullRequests = pullRequestAmount.size();
+		final List<PullRequest> results = new ArrayList<>( numPullRequests );
+		for ( int i = 0; i < numPullRequests; i++ ) {
+			final int id = pullRequestAmount.get( i );
+			final String source = jsonPath.read( "$.[" + i + "].head.ref" );
+			final String destination = jsonPath.read( "$.[" + i + "].base.ref" );
+			final String lastUpdate = jsonPath.read( "$.[" + i + "].updated_at" );
+			results.add( PullRequest.builder() //
+					.id( id ) //
+					.source( source ) //
+					.destination( destination ) //
+					.lastUpdate( lastUpdate ) //
+					.build() ); //
+		}
+		return results;
+	}
+
+	private static String requestPath( final PullRequest pullRequest ) {
+		return "/pulls/" + pullRequest.getId();
+	}
+
+	private DocumentContext jsonPathForPath( final String urlPath ) {
+		final String json = template.getForObject( urlPath, String.class );
+		return JsonPath.parse( json );
+	}
+
+	@Override
+	public void addComment( final PullRequest pullRequest ) {
+		final Map<String, String> request = new HashMap<>();
+		request.put( "body", "This pull request needs some manual love ..." );
```

Comment:
Should the message be a parameter for generic use?

Reply:
Good point, fixed